### ISSUE TITLE
AWS access for Aaron Hietpas

### ISSUE THREAD

**Issue by mviterna** [department-of-veterans-affairs/va.gov-team#48688]
### COR Name

Cecila Lee

### Vendor Onboarding Representative Name

Liberty IT -  Michael G. Lovejoy

### Your Name

Aaron Hietpas

### Your Email

 Hietpas_Aaron@bah.com

### Team, Role, and Company of the target individual

Ninja Pigs

### Product Manager (PM) name and email

Nichole Harris - Harris_Nichole@bah.com

### Product Owner (PO) name and email

Janet Coutinho janet.coutinho@va.gov

### Desired AWS Access

Aaron needs similar access to Mark Viterna. This includes console access, S3, etc. Aaron is not on the VFS roster. This appears to be a mistake, he has been a developer since before the Liberty/BAH acquisition and has SOCKS access and has been vetted/onboarded. Can we also get him added to the roster?

### Access Expiration

ongoing

### Additional Notes

Aaron needs this as soon as possible to provide production support for the Forms API in Lighthouse. Currently no one on the team has access.

### User must exist in a roster before AWS access can be granted

- [ ] Search for user on the VFS Team Roster: https://docs.google.com/spreadsheets/d/11dpCJjhs007uC6CWJI6djy3OAvjB8rHB65m0Yj8HXIw/edit?folder=0ALlyxurHpUilUk9PVA#gid=2042046665
- [ ] Or search for user on the Platform Team Roster: https://vfs.atlassian.net/wiki/spaces/AP/pages/1908834623/Platform+Roster
- [x] If user is on a VFS team but not in the VFS Team Roster, add the 'NOT YET' label and instruct them to start the Platform orientation process https://depo-platform-documentation.scrollhelp.site/getting-started/Platform-Orientation.1877344532.html
- [ ] If a user is on a Platform team but not on the Platform Team Roster in Confluence, add the 'NOT YET' label and instruct them to reach out to their Product Manager to be added.
- [ ] Comment in this issue saying which roster the user is listed in.

**Comment by ianhundere** [comment]
@aaronhietpas please start the Platform orientation process https://depo-platform-documentation.scrollhelp.site/getting-started/Platform-Orientation.1877344532.html

**Comment by mviterna** [comment]
@ianhundere Aaron has been on the Lighthouse platform for awhile now. It appears he just was never added to the roster. Can we just get him added? I believe he came on board before this new requirement, but was just not added to the roster.

**Comment by ianhundere** [comment]
@mviterna you'll need to reach out to their PM to do that.

**Comment by mviterna** [comment]
@ianhundere He should be added now

**Comment by td-usds** [comment]
@mviterna: can you elaborate on why membership of `adhoc-vetsgov-ssm-prod` would be required for Aaron?

**Comment by td-usds** [comment]
@aaronhietpas can you elaborate on https://github.com/department-of-veterans-affairs/va.gov-team/issues/48688#issuecomment-1318747971

**Comment by td-usds** [comment]
@aaronhietpas ping?

**Comment by aaronhietpas** [comment]
@td-usds I am on the engineering team supporting the VA Forms, Veteran Verification and Veteran Confirmation APIs that are running on the Vets-API Platform. No one else on our team currently has access to view the rails console of these APIs in production which can hinder our ability to troubleshoot any issues/answer any questions that come up from consumers.

**Comment by BillChapmanUSDS** [comment]
@aaronhietpas   I'm taking over this review from Thomas and this is my first week reviewing it. You'll have to bear with me here as I sort out our process. I'm just getting up to speed on how generous/restrictive I can be.  If anyone else on this thread wants to chime in, please do. 
 
For what it's worth, I can think of plenty of reasons you may want to jump to the Rails console on production, I've certainly done it myself more than I should admit, and it makes some things a lot easier. So, I really do get it. 
 
However, I'm not sure it's a practice we should be embracing at an organization of this size.  It may be a nicety of the Rails framework but there are plenty of stacks that do not have an app environment aware REPL available and operations teams firefight those just fine. Also, it grants way more access and liability than an engineer should reasonably want. I'd prefer if we were not instantiating models on production data in order to troubleshoot a bug.   
 
With that said:  I think I'm **ok granting the typical temporary access (3 months)** but I'd want to see a plan for mitigating the reasons we are doing it in the first place. 

TL;DR;  ** If you or someone else can give me some brief answers to the questions below so I can help find solutions to what is missing I'll OK this access.  but I'll probably want to have a deeper conversation before the next time we renew (I'll be more up to speed by then)** 
 
1.  Do we not have adequate ability to reproduce bugs in a lower environment? 
2. Are you missing the ability to access productions logs and real time metrics suffcien't for firefighting? 
3.  Can you give me an example of the last time there was a need to log in to production and instantiate things in the Rails console and why?

**Comment by little-oddball** [comment]
@aaronhietpas - Any update(s)?

**Comment by BillChapmanUSDS** [comment]
Approved for 3 months. 

Approved for 3 months As per some recent discussions about elevated privileges I'm going to ok this for 3 months as wwe get more detail about why elevated privileges are so often needed and how to mitigate

**Comment by rmtolmach** [comment]
Since this request is set for three months, access expiration needs to be set up before this can be closed. See Access Expiration section here: https://vfs.atlassian.net/wiki/spaces/OT/pages/1792114735/Onboard+New+Team+Members+Granting+Access+to+Internal+Tools#Access-Expiration

**Comment by oseasmoran73** [comment]
[Reminder](https://dsva.slack.com/archives/C02RLAUM2LV/p1678295573873449) can be found in tech-support channel. Closing ticket

**Comment by rmtolmach** [comment]
@aaronhietpas You prod access is expiring. Do you still need the Prod access or can we remove it?

**Comment by aaronhietpas** [comment]
Yeah you can remove it

**Comment by rmtolmach** [comment]
Thanks! ✅ done in https://github.com/department-of-veterans-affairs/devops/pull/13076
